### ISSUE TITLE
No Popup, even though battery % can be read inside the app correctly.

### ISSUE THREAD

**Issue by Craniniaxx** [steam3d/MagicPods-Windows#405]
So I have recently bought the MagicPods app, mainly due to the sheer amount of positive reviews it holds and it did not disappoint me at the first glance. But even during the setup, I noticed that there was no pop-up for the AirPods. But thought it might be disabled. Went back, checked, it was enabled. Searched for it on the web, **did the slider search radius thingy close to a hundred times, with all the different methods**, closing with each progression etc. still no popup. Thought it might be my device's issue or incompatibility with the UI, so turned on the "Sound input pop-up" _AND THAT WAS WORKING JUST FINE!!_ Then I went to the official website, saw the **MagicAAP**, installed it. Opened app again after restarting, _nearly punched myself in the face_. Now it was showing accurate battery for the both L&R Buds as well as the Case... **BUT STILL NO POPUP**... Can anyone PLEASE help me on this... At this point I'm just sad! Here is an attached picture of the Interface... 
![17257737552315276384515990979112](https://github.com/user-attachments/assets/1ed8adb5-ae61-41d3-ac29-0b7e59528db7)

**Comment by steam3d** [comment]
Hi. Update Bluetooth driver and check system files 
https://support.microsoft.com/en-us/windows/using-system-file-checker-in-windows-365e0031-36b1-6031-f804-8fd86e0ef4ca

Possible Bluetooth LE doesn’t work and MagicPods can't find nervy devices.

**Comment by Craniniaxx** [comment]
Thank you for your reply, I'll do that and share the results. Could you tell me what I should be looking for?

**Comment by steam3d** [comment]
This is an unknown issue. Some users have the same issue, but we still haven't figured out the solution. For some users, reinstalling the app helps, for others updating the Bluetooth driver helps. I have some feedback from users that reinstalling Windows and trying the MagicPods again worked for them.

Because I didn't change anything on MagicPods, it means that some Windows setting is blocking the use of Bluetooth LE.

**Comment by Craniniaxx** [comment]
I see... I'll try the other two methods, can't reinstall the windows due to a reason. Right now I'm doing the DISM check as you guided me. Will get back after that. Also, I should add, the app seems to be working Perfectly-Fine, other than the pop up. All settings work, the sound is excellent, no drop in connectivity, live tiles are okay, the tray icons are accurately showing the data, every thing is shown without any bug, haven't experienced any crashes, and the Pop-up for when you change the audio source is working perfectly fine as well. Its JUST the pop-up which is not appearing. The device connects with the Buds as soon as I open the case, but due to some unknown reason just doesn't show the Pop-up. :/

**Comment by Craniniaxx** [comment]
Also, could you please point me towards how can I update my bluetooth driver. I'm in test mode currently so doesn't matter if the driver is officiall or not. I'm not very tech-oriented and have very low knowledge on the part, it would be a great help. Because I think what you said might be the actual issue, I just got in my device manager and under the Bluetooth driver, which reads, **Qualcomm QCA9565 Bluetooth 4.0** under the properties, the driver date is 12/22/2017! 😐

**Comment by steam3d** [comment]
Possibly this is the issue. Run [Bluetooth Device Finder](https://magicpods.app/utils/bdf.zip) for 5 seconds and upload the log.txt file that will appear near the exe file here.

**Comment by Craniniaxx** [comment]
[log.txt](https://github.com/user-attachments/files/16921315/log.txt)

**Comment by steam3d** [comment]
It's definitely Bluetooth issue. Go to manufacture site, download and install the Qualcomm QCA9565 Bluetooth 4.0 driver

**Comment by Craniniaxx** [comment]
Thanks, I'll search for it as soon as the DISM is over.

**Comment by Craniniaxx** [comment]
Here are the logs after completion of the checks that you advised... 

[CBS.log](https://github.com/user-attachments/files/16921700/CBS.log)

**Comment by Craniniaxx** [comment]
UPDATE: I updated my drivers, now they stand at their latest updates. They also support the LE. Below attached is the log of the software that you sent me taken after the update and restart. **Bluetooth Device Finder** 

I have uninstalled the app because the pop-up was still not showing, I'll see if reinstalling will make a difference or not. Thank you for staying with me btw.. :)

[log.txt](https://github.com/user-attachments/files/16921882/log.txt)

**Comment by Craniniaxx** [comment]
No change even after reinstalling. I don't know what to do now... Please guide me if you can.

**Comment by steam3d** [comment]
Possibly, it's a Bluetooth hardware issue. The MaxAdvertisementDataLength is 31 in you Bluetoothspecifications. I recall some users having the same issue. They replaced their Bluetooth dongle.

In your case, the popup will not work. I cannot do anything from my side because the MagicPods do not receive any data from Bluetooth LE.

**Comment by Craniniaxx** [comment]
Thank you so much for your assistance. I will try seeing what I can do about that. The app itself is great though.

**Comment by Craniniaxx** [comment]
I've got to add that I've read online during looking for the solution & the MAXADVERTISEMENTDATALENGTH has always a universal limit which is 31 bytes per advertising data and another 31 bytes for scan response. So that cannot be changed even if I change the hardware. It is something else that is blocking it. I'll keep looking for a solution and post it here when I find the solution.

**Comment by steam3d** [comment]
My Bluetooth:

| Name                                   | Value |
| -------------------------------------- | ----- |
| AreClassicSecureConnectionsSupported   | True  |
| AreLowEnergySecureConnectionsSupported | True  |
| IsAdvertisementOffloadSupported        | True  |
| IsCentralRoleSupported                 | True  |
| IsClassicSupported                     | True  |
| IsExtendedAdvertisingSupported         | True  |
| IsLowEnergySupported                   | True  |
| IsPeripheralRoleSupported              | True  |
| MaxAdvertisementDataLength             | 160 | 

Your Bluetooth:

| Name                                   | Value |
| -------------------------------------- | ----- |
| AreClassicSecureConnectionsSupported   | False |
| AreLowEnergySecureConnectionsSupported | True  |
| IsAdvertisementOffloadSupported        | False  |
| IsCentralRoleSupported                 | True  |
| IsClassicSupported                     | True  |
| IsExtendedAdvertisingSupported         | False |
| IsLowEnergySupported                   | True  |
| IsPeripheralRoleSupported              | False  |
| MaxAdvertisementDataLength             | 31 | 

I think this is something connected with Bluetooth dongle.

**Comment by steam3d** [comment]
Run this app and open AirPods case near your PC. Upload the log.txt here. I changed code, maybe we get some other result.
[bt_test.zip](https://github.com/user-attachments/files/16927663/bt_test.zip)

**Comment by Craniniaxx** [comment]
Surely, by all means... 
here is the log file 
[log.txt](https://github.com/user-attachments/files/16942750/log.txt)

**Comment by steam3d** [comment]
Well, it works now, To find out exactly the cause of the problem with Bluetooth, repeat the same with [Bluetooth Device Finder](https://magicpods.app/utils/bdf.zip)

Do you have Fake AirPods Pro 2? Because the log.txt contains data from Fake AirPods Pro 2

**Comment by Craniniaxx** [comment]
Nope, not as far as I know...  Could you please tell me more?
Here is the log
[log.txt](https://github.com/user-attachments/files/16943043/log.txt)

**Comment by steam3d** [comment]
Did you change anything with Bluetooth? Because today, both versions of app is work.

1. Fake AirPods is always put 00 at the end of the data.Here is a sample of data from your log.txt
```
07-13-01-14-20-75-AA-37-30-00-10-00-E4-E4-5E-00-00-00-00-00-00
```

2. Windows Bluetooth devices list contains AirPods Pro Gen 1, but the data from your log.txt from AirPods Pro Gen 2.

3. Your AirPods mac address is 60:BB:E7:29:5D:5C.  Here is the result of checking it:
https://macaddress.io/mac-address-lookup/r2ebG4qm5x
![изображение](https://github.com/user-attachments/assets/71d7adcb-6e75-4522-9d79-1d2f39b500c9)

Vendor field must be **Apple**

Connect AirPods Pro 2 to Windows. Open Settings (Win+I) go to Bluetooth and find your connected AirPods. Does battery level display there?

**Comment by Craniniaxx** [comment]
Yes, I just updated the drivers... nothing more than that. and yeah, i checked, the battery level is not being shown in the windows bluetooth setting... does that mean they are fake :(

**Comment by Craniniaxx** [comment]
Ahh... I see.... So you found the problem. Is there a solution for this??

**Comment by steam3d** [comment]
No, it's unexpected behavior from the AirPods. I recommend checking your AirPods; I'd love to be wrong, but they really look like fake ones.

Anyway, I see that MagicPods can control AirPods using MagicAAP, so at least you can use it without the popup window.

**Comment by Craniniaxx** [comment]
So no pop-up... Alas! Well after reading this, I did check the serial number and also updaded the bt drivers one more time. After pairing them again, they are being recognized as Pro 2 instead of just Pro.. Checked the device manager too... 
Here is the picture. I had checked them originally with an iPhone, the ANC was working perfectly as well as the sound was also amazingly good, the seller took full price from me so was under the impression that these are original. That's some devastating news that you gave me! 
![IMG_20240910_183131.jpg](https://github.com/user-attachments/assets/4d8a61f6-b91f-4d6b-bbcc-8c51175db86a)

**Comment by Craniniaxx** [comment]
But really man, at this point, the amount of ease and functionality alone your application provides, I've been using it for a lil time now... It doesn't even matter that the pop-up doesn't appear. They connect as soon as I open the case, I can see the battery of individual pods in the taskbar, can cycle through the sound modes, enable or disable handsfree mode, enable low latency mode (which is a life saver as I am an editor)... and to top it off, I'll say that thank you for being so helpful and responding this fast. You're really a good developer. Keep up the good work, Man!
